the-collab-lab/tcl-28-smart-shopping-list
# Isa - reusable button component

## Description

- Created a reusable `GreenButton` 🟢 component that can be used throughout the application (however, this button component does not replace the delete or search buttons)
- Altered styling of button in `src/themes/theme.js` so that there is a border radius
- Imported the button component into the following components:
    - `Home`
    - `AddItem`
    - `ItemsList`
- Replaced existing buttons in those components (with exception to delete and clear search buttons) with GreenButton component
- Passed props to the component: 
    - `btnText` - this prop represents the button text
    - `clickFunction` - this prop represents whatever function runs whenever the button is clicked, like a `handleClick` or `handleSubmit` for example
 - Used the `GreenButton` component in the `ItemsList` & `AddItem` components as a way to navigate to other pages. The existing navigation links we had at the bottom of the page are still there just in case we want to revert back to those or do something different
 - Created a /styles folder inside source and a `GreenButton.css` file for styling, but that file is kind of useless tbh 😅

## Related Issue
Closes #45 

## Acceptance Criteria

Create a reusable button component that will be used throughout the application

AC

- Set background color to #98D79A
- Add border-radius styling
- Ensure that the button can be used to:
    - create a new token
    - submit existing token
    - add a new item
    - exit the list
    - navigate to add item or view list page (maybe this will be for another issue?)

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/56979810/131416587-4d9c8205-e0cc-4922-97ff-a8cead757de4.png)
![image](https://user-images.githubusercontent.com/56979810/131416616-ebadee5d-334b-41ab-9f82-28dbda55d3d7.png)

### After

![image](https://user-images.githubusercontent.com/56979810/131416343-191b952d-0ffe-49f7-967e-c9e8cb1a38cd.png)
![image](https://user-images.githubusercontent.com/56979810/131416408-24859f81-69dc-430e-b835-08acd6d5505a.png)
![image](https://user-images.githubusercontent.com/56979810/131416440-70acb19f-fb1c-4c1c-bf0c-6eed62e088cf.png)
![image](https://user-images.githubusercontent.com/56979810/131416497-b06fe286-9529-41ba-bf36-399273824332.png)

## Testing Steps / QA Criteria

- `git checkout ia-button-component`
- `npm start`
- Test out the buttons to make sure they still function like they're supposed to. I think there's a bit of a lag when you try to join an existing list for some reason
